scipy/misc/doccer.py

```diff
@@ -18,12 +17,29 @@ def __dir__():
 def __getattr__(name):
     if name not in __all__:
         raise AttributeError(
-            "scipy.misc.doccer is deprecated and has no attribute "
-            f"{name}.")
-
-    warnings.warn("The `scipy.misc.doccer` namespace is deprecated and "
-                  "will be removed in SciPy v2.0.0.",
-                  category=DeprecationWarning, stacklevel=2)
-
-    return getattr(doccer, name)
-
+            f"`scipy.misc.doccer` has no attribute `{name}`; furthermore, "
+            f"`scipy.misc.doccer` is deprecated and will be removed in SciPy 2.0.0."
+        )
+
+    attr = getattr(import_module("scipy._lib.doccer"), name, None)
+
+    if attr is not None:
+        message = (
+            f"Please import `{name}` from the `scipy._lib.doccer` namespace; "
+            f"the `scipy.misc.doccer` namespace is deprecated and "
+            f"will be removed in SciPy 2.0.0."
+        )
+    else:
+        message = (
+            f"`scipy.misc.doccer.{name}` is deprecated along with "
+            f"the `scipy.misc.doccer` namespace. "
+            f"`scipy.misc.doccer.{name}` will be removed in SciPy 1.13.0, and "
+            f"the `scipy.misc.doccer` namespace will be removed in SciPy 2.0.0."
+        )
+
+    warnings.warn(message, category=DeprecationWarning, stacklevel=3)
```

Comment:
I think this might be stacklevel 2 instead because the helper function introduces another stack frame which we don't have here

Reply:
```suggestion
    warnings.warn(message, category=DeprecationWarning, stacklevel=2)
```